Update mobile app component on the jetpack assistant summary

CodeyGuyDylan opened the pull request:
Update the mobile component on the assistant summary page and add contexts in which it shows up.

#### Changes proposed in this Pull Request:

This adds functionality that detects the user's device when this page is opened, and displays the correct form of link to the App depending on the device. If the device is apple, an App Store link will show up, if it is Android, the Google Play store link will show up. Otherwise, a QR code will show up.

Additionally, this mobile component used to only show up if users had Jetpack Security. Now, if users only have backup, this component will show up underneath the section about `one-click-restores` or managing security. It will also show up underneath the summary section if users do not have Security or Backup.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

P2: pbNhbs-3od-p2

#### Does this pull request change what data or activity we track or use?
A link that used to be tracked that went to https://wordpress.com/ has been changed to https://jetpack.com/mobile and is tracked when clicked.

#### Testing instructions

This is a decently large change, so there are two separate testing plans. One to conditionally show a QR code/app link, and then one to test out the new contexts of the component

**Testing the conditionally rendered QR code/app links**

1. Check out this branch via `git switch update/mobile-app-component-to-assistant-summary`
2. Get your local environment up and running
3. Head over to `/wp-admin/admin.php?page=jetpack#/recommendations/summary`. No matter your setup, the mobile app section should be visible in some form
4. Assuming you are doing this all on a desktop (not sure how you would do this otherwise), the QR code should show up in the component
![Screen Shot 2022-09-07 at 1 42 24 PM](https://user-images.githubusercontent.com/65001528/188963292-7c61c2c2-1a1a-4537-9c60-46331bc12ee3.jpg)
5. Open up the Chrome Dev tools (I am not sure how to do this in other browsers, but I am sure there is a way) and find the 'Network Conditions' tool
![image](https://user-images.githubusercontent.com/65001528/188963652-7d240168-ef63-40d1-811d-fb61f3d85c7b.png)
6. In the panel, unselect the 'Use browser default' under User Agent, and select `Chrome - iPad`
![image](https://user-images.githubusercontent.com/65001528/188963875-a0fbc9b7-537f-4fb4-8433-299b88a90a36.png)
7. Refresh the page, and you should now see the App Store button
![image](https://user-images.githubusercontent.com/65001528/188964007-c7ae9561-61b0-4e5b-bde2-d9a3c13842b8.png)
8. Back in the `Network Conditions` panel, select `Chrome - Android Mobile`
![image](https://user-images.githubusercontent.com/65001528/188964135-cb82495e-ffa0-419b-9547-cdee2e486a06.png)
9. You should now see the Google Play store button
![image](https://user-images.githubusercontent.com/65001528/188964186-20493b64-5e9b-4dfc-8253-30859f37ebed.png)

If you want more devices to test on, you can use BrowserStack. Automattic has accounts to use so you don't have to pay for it. More info here: PCYsg-iPs-p2 and here: 2d7c8-pb

I have tested this on Apple phones, tablets, and desktop; Android phones and tablets, and Windows phones/desktop

**Testing new contexts in which mobile component shows up**

For this test, I would advice removing all subscriptions to jetpack from your local environment for simplicities sake, but this should work just find as long as you don't have JP Security or Backup

Figma copy: RvAStEukGsx3Tfw0rB92p0-fi-23721%3A6029

1. For the first test, ensure you do not have Jetpack Security or Jetpack Backup on the site
2. Head to `/wp-admin/admin.php?page=jetpack#/recommendations/summary` 
3. You should see the mobile component on the bottom of the screen underneath the summary
![Screen Shot 2022-09-07 at 1 57 29 PM](https://user-images.githubusercontent.com/65001528/188965767-60057ecf-96bf-426e-a6bb-763830356221.jpg)
4. Make sure you see the mobile component and check the figma copy to make sure the text matches what it is supposed to be. Also play with the view width to make sure it looks good on mobile sized screens.
5. Next, go buy a month of Jetpack Backup with credits and head back to the assistant summary screen.
6. You should see the mobile component with different text displayed underneath the `one-click-restores` component. Test to make sure it matches the figma mockup and looks good on mobile
![Screen Shot 2022-09-07 at 2 03 17 PM](https://user-images.githubusercontent.com/65001528/188966764-a9d26a80-f8db-438f-b6e8-e58d4e852e73.jpg)
7. For testing the component when one click restores are enabled, I was not able to figure out how to get those enabled on my development environment. So to more easily test this out, go to the file in the code located at `projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx` and on line 119 replace `sidebarCardsSlug` with `'manage-security'` and do the same on line 141. It will look like this
```javascript
if ( isFetchingSidebarData ) {
		sidebarCards = <JetpackLoadingIcon altText={ __( 'Loading recommendations', 'jetpack' ) } />;
	} else {
		switch ( 'manage-security' ) {
			case 'loading':
				sidebarCards = (
					<JetpackLoadingIcon altText={ __( 'Loading recommendations', 'jetpack' ) } />
				);
				break;
			case 'upsell':
				sidebarCards = upsell.hide_upsell ? <ProductCardUpsellNoPrice /> : <SummaryUpsell />;
				undersideCards = <MobileApp slug={ sidebarCardsSlug } />;
				break;
			case 'one-click-restores':
				sidebarCards = (
					<>
						<OneClickRestores />
						<MobileApp slug={ sidebarCardsSlug } />
					</>
				);
				break;
			case 'manage-security':
				sidebarCards = (
					<>
						<Security />
						<MobileApp slug={ 'manage-security' } />
					</>
				);
				break;
			case 'download-app':
				sidebarCards = <MobileApp slug={ sidebarCardsSlug } />;
				break;
			default:
				sidebarCards = <MobileApp slug={ 'unknown' } />;
		}
	}
```

8. Head back to the assistant summary page and you should see the mobile component below the manage security component. Make sure it matches the figma mockup and looks good on mobile
![Screen Shot 2022-09-07 at 2 06 37 PM](https://user-images.githubusercontent.com/65001528/188967459-359a2b08-c2bb-4406-b9ce-49bed185e5c6.jpg)
9. Lastly, buy one month of security with credits ( you may have to remove your backup plan )
10. Head back to the assistant summary page and you should see the mobile component alone in the sidebar. Make sure it matches the figma file and looks good on mobile sizes
![Screen Shot 2022-09-07 at 2 09 48 PM](https://user-images.githubusercontent.com/65001528/188968026-5e330bcd-074f-46de-83fd-ba6a306830a2.jpg)

Man that was a lot of testing instructions! Thanks for looking it all over, whoever you may be.

Commit:
[not verified] Add new images for mobile sidebar card header

Commit:
[not verified] Update mobile sidebar card header

Commit:
[not verified] Change heading text

Commit:
[not verified] Add dynamic body text and update layout of sidebar

Commit:
[not verified] Add device detecting function

Commit:
[not verified] Rename function and add README

Commit:
[not verified] Add conditionally rendered app links

Commit:
[not verified] make isUnderside future proof

Commit:
[not verified] Add conditional styling for when card is under the summary

Commit:
[not verified] changelog

CodeyGuyDylan commented on projects/plugins/jetpack/_inc/client/recommendations/sidebar/mobile-app/index.tsx:
```diff
@@ -0,0 +1,201 @@
+import { imagePath } from 'constants/urls';
```
I converted this file to TypeScript because we seem to be moving in that direction. Almost nothing is the same from the previous version of the file, so don't worry about looking at the deleted `jsx` file to compare

CodeyGuyDylan commented on projects/plugins/jetpack/_inc/client/recommendations/summary/style.scss:
```diff
@@ -14,7 +14,7 @@
 	border: 1px solid $gray-lighten-20;
 
-	@include breakpoint( '<660px' ) {
+	@include breakpoint( '<782px' ) {
```
Layout was somewhat broken between 660 and 782 so I updated it to switch to the column view earlier

Commit:
[not verified] Add QR codes with unique utm_source for each context

Commit:
[not verified] Change isUnderside into a prop

Commit:
[not verified] Remove test code

monsieur-z commented on projects/plugins/jetpack/_inc/client/scss/variables/_colors.scss:
```diff
@@ -37,6 +37,9 @@ $gray-text-min: darken( $gray, 18% ); //#537994
 // See wordpress.com/design-handbook/colors/ for more info.
 
+// Jetpack colors
+$jetpack-green-50: #008710;
```
I think you could use `$studio-jetpack-green-50`, defined in `projects/plugins/jetpack/modules/calypsoify/_studio-wpcom.scss`.

CodeyGuyDylan replied:
Looks like I didn't actually end up using this, I will remove it

monsieur-z commented on projects/plugins/jetpack/_inc/client/recommendations/sidebar/mobile-app/index.tsx:
```diff
@@ -0,0 +1,195 @@
+				height={ headerImgSize }
+				alt="wordpress icon"
+			/>
+			{ /* Extra 2 pixels on width and height are to account for the white padding on this SVG, so the
```
Is it possible to edit the SVG instead?

CodeyGuyDylan replied:
Yeah not sure why I didn't think of that 😅

monsieur-z commented on projects/plugins/jetpack/_inc/client/recommendations/sidebar/mobile-app/index.tsx:
```diff
@@ -0,0 +1,195 @@
+				src={ imagePath + '/recommendations/wordpress-icon.svg' }
+				width={ headerImgSize }
+				height={ headerImgSize }
+				alt="wordpress icon"
```
Since icons are decorative here, I think it'd be a better experience for screen reader users to have an empty `alt` attribute.

CodeyGuyDylan replied:
Agreed 😄

monsieur-z reviewed (dismissed):
This all looks good to me. I left some minor comments in the code.

Regarding the UI, a nitpick could be to center the card vertically in the last use case, to conform to the mockups.
<img width="1010" alt="Screen Shot 2022-09-20 at 1 00 11 PM" src="https://user-images.githubusercontent.com/1620183/191322745-7d7929d5-8a6c-48d7-9a06-145543e68161.png">

Commit:
[not verified] Address small issues

Remove unused color that was added
Edit SVG to not include 1px of padding
Remove alt text to improve screen reader experience

CodeyGuyDylan commented:
@monsieur-z I addressed the small issues you brought up, not sure if you want to re-review or not, but I'll add the tag `Needs Review`

monsieur-z reviewed (dismissed):
@CodeyGuyDylan, I've checked your latest commit, which looks good to me. I haven't gone through the test plan again, since updates are pretty straightforward.

bindlegirl commented:
I tested the App Store badge on my iPhone and I get redirected to the WP app in the App Store instead of the Jetpack one.

CodeyGuyDylan commented:
> I tested the App Store badge on my iPhone and I get redirected to the WP app in the App Store instead of the Jetpack one.

Thank you for catching this, I will fix this and make sure the android link works too

bindlegirl commented:
I checked. Android needs fixing too :-)

Commit:
[not verified] Update links to link to the Jetpack app

CodeyGuyDylan commented:
@bindlegirl thank you for that, it should be fixed now! I updated the AppsBadge component directly because it is only used here and in the AppsCard component, but the AppsCard component uses hardcoded links so it should not change this anywhere else

samiff committed:
Update README.md

samiff committed:
Merge branch 'trunk' into update/mobile-app-component-to-assistant-summary

bindlegirl reviewed (approved):
LGTM! tested on JN site and mobile devices. Thanks for detailed instructions :-)

samiff reviewed (approved):
Updated app links worked well for me 👍 

Not related to this PR, but the app badges do get a bit hard to read when resized:

![Screen Shot 2022-09-26 at 11 36 40 AM](https://user-images.githubusercontent.com/15803018/192344737-68cd24f6-a9ee-41e2-89b4-55527f8a700c.png)

Waiting on API limit for the checks to rerun, internal: p1664213101065879-slack-CBG1CP4EN

CodeyGuyDylan commented:
> Updated app links worked well for me 👍
> 
> Not related to this PR, but the app badges do get a bit hard to read when resized:
> 
> ![Screen Shot 2022-09-26 at 11 36 40 AM](https://user-images.githubusercontent.com/15803018/192344737-68cd24f6-a9ee-41e2-89b4-55527f8a700c.png)
> 
> Waiting on API limit for the checks to rerun, internal: p1664213101065879-slack-CBG1CP4EN

Thank you for the slack link, was wondering what was going on.

As far as the sizing goes, I'll put that in our maintenance backlog and see what we want to do about that 😄

CodeyGuyDylan commented:
The plus side about the badge size is that in this context it should only show up on mobile, in which I believe it looks a bit more visible, but I agree we may want to address the instances where it shows up on desktop in other cases that exist

CodeyGuyDylan committed:
Merge branch 'trunk' into update/mobile-app-component-to-assistant-summary

CodeyGuyDylan committed:
Merge branch 'trunk' into update/mobile-app-component-to-assistant-summary

CodeyGuyDylan committed:
Merge branch 'trunk' into update/mobile-app-component-to-assistant-summary
